core/src/main/scala/shapeless/typeoperators.scala

```diff
@@ -201,3 +201,41 @@ object Lub {
   }
 }
 
+/**
+ * Type class witnessing the implicit refinement type that extends `TypeClassConstraint`.
+ *
+ * @note `WitnessThe` is similar to [[the]], except `WitnessThe` can be used in another implicit type class resolution process, e.g. `DependentLiftAll`.
+ *
+ * @author Yang Bo
+ */
+trait WitnessThe[TypeClassConstraint] extends DepFn0 {
+
+  type Out <: TypeClassConstraint
+
+}
+
+object WitnessThe {
+
+  type Aux[TypeClassConstraint, Out0] =
+    WitnessThe[TypeClassConstraint] {
+      type Out = Out0
+    }
+
+  final class WitnessTheAux[TypeClassConstraint, Out0 <: TypeClassConstraint](override val apply: Out0)
+      extends WitnessThe[TypeClassConstraint] {
+    type Out = Out0
+  }
+
+  @macrocompat.bundle
+  private[WitnessThe] final class Macros(val c: whitebox.Context) {
+    import c.universe._
+    def apply[TypeClassConstraint: WeakTypeTag](typeClass: Tree): Tree = {
+      q"new _root_.shapeless.WitnessThe.WitnessTheAux[${weakTypeOf[TypeClassConstraint]}, ${typeClass.tpe}]($typeClass): _root_.shapeless.WitnessThe.Aux[${weakTypeOf[TypeClassConstraint]}, ${typeClass.tpe}]"
+    }
+  }
+
+  implicit def apply[TypeClassConstraint](implicit typeClass: TypeClassConstraint)
+    : WitnessThe.Aux[TypeClassConstraint, /* typeClass.type */ _ <: TypeClassConstraint] =
```

Comment:
The real type is `WitnessThe.Aux[TypeClassConstraint, typeClass.type]`.

 `WitnessThe.Aux[TypeClassConstraint, typeClass.type]` is not a subtype of `WitnessThe.Aux[TypeClassConstraint, TypeClassConstraint]`